ext/node/polyfills/http.ts

```diff
@@ -341,7 +341,7 @@ class ClientRequest extends NodeWritable {
       path = "/" + path;
     }
     return `${protocol}//${auth ? `${auth}@` : ""}${host}${
-      port === 80 ? "" : `:${port}`
+      (port === 80 || opts.host) ? "" : `:${port}`
```

Comment:
**question:** Is it possible, and does Node work with a configuration where host and port are both set and the host does not define a port? ie. Should we check for `:` presence in `opts.host` here?